packages/core/src/hooks/useContract.ts

```diff
@@ -1,30 +1,72 @@
+import { Abi, TypedContract } from "abi-wan-kanabi";
+import {
+  ExtractAbiFunctionNames,
+  FunctionArgs,
+  FunctionRet,
+} from "abi-wan-kanabi/dist/kanabi";
 import { useMemo } from "react";
-import { Abi, Contract, ProviderInterface } from "starknet";
+import { CallOptions, Contract, ProviderInterface } from "starknet";
 
 import { useStarknet } from "~/context/starknet";
 
+// did this because "Omit" wont work directly over an abstract class
+type _Contract = {
+  [K in keyof Contract]: Contract[K];
+};
+
+export type StarknetTypedContract<TAbi extends Abi> = TypedContract<TAbi> &
+  Omit<_Contract, "populate" | "populateTransaction" | "call"> & {
+    call<TFunctionName extends ExtractAbiFunctionNames<TAbi>>(
+      method: TFunctionName,
+      args?: FunctionArgs<TAbi, TFunctionName>,
+      options?: CallOptions
```

Comment:
defined `call` type explicitly as `abi-wan-kanabi` didnt have options parameter as third argument but `starknet.js` did

Reply:
Where is `call` used? In `useCall` you cast to `Contract` so I don't think it's needed there right?

Reply:
> Where is `call` used? In `useCall` you cast to `Contract` so I don't think it's needed there right?

it's not used anywhere else except in `useCall`  which as you mentioned right that its casted as `Contract` so its of no use for us, but i added it assuming if user ever uses `useContract` directly, and uses its `call` so they can have strict type safety  along with the `options` parameter.
let me know your thoughts, we can remove it if you think its not needed.